geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKey.java

```diff
@@ -29,41 +28,23 @@
 import org.apache.geode.internal.serialization.DeserializationContext;
 import org.apache.geode.internal.serialization.KnownVersion;
 import org.apache.geode.internal.serialization.SerializationContext;
-import org.apache.geode.redis.internal.executor.cluster.CRC16;
 import org.apache.geode.redis.internal.executor.cluster.RedisPartitionResolver;
 
 public class RedisKey implements DataSerializableFixedID {
 
-  private int crc16;
+  private int slot;
   private byte[] value;
 
   public RedisKey() {}
 
   public RedisKey(byte[] value) {
     this.value = value;
-    int startHashtag = Integer.MAX_VALUE;
-    int endHashtag = 0;
-
-    for (int i = 0; i < value.length; i++) {
-      if (value[i] == '{' && startHashtag == Integer.MAX_VALUE) {
-        startHashtag = i;
-      } else if (value[i] == '}') {
-        endHashtag = i;
-        break;
-      }
-    }
-
-    if (endHashtag - startHashtag <= 1) {
-      startHashtag = -1;
-      endHashtag = value.length;
-    }
-
-    crc16 = CRC16.calculate(value, startHashtag + 1, endHashtag);
+    slot = KeyHashUtil.slotForKey(value);
   }
 
   public int getBucketId() {
     // & (REDIS_SLOTS - 1) is equivalent to % REDIS_SLOTS but supposedly faster
```

Comment:
remove this comment since it no longer does & (REDIS_SLOTS - 1)

Reply:
done